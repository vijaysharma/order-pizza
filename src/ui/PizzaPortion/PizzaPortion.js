import React from 'react';
import PropTypes from 'prop-types';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faPizzaSlice} from '@fortawesome/free-solid-svg-icons';

import * as classes from './PizzaPortion.module.css';

const PizzaPortion = props => {
  let sliceClass = '';
  switch (props.size) {
    case 'small':
      sliceClass = classes.SmallSlice;
      break;
    case 'medium':
      sliceClass = classes.MediumSlice;
      break;
    case 'large':
      sliceClass = classes.LargeSlice;
      break;
    default:
      sliceClass = '';
  }
  const label = props.label ? props.label.toUpperCase() : '';
  return (
    <div className={classes.PizzaSlice}>
      <FontAwesomeIcon icon={faPizzaSlice} className={sliceClass} />
      <span>{label}</span>
    </div>
  );
};
PizzaPortion.defaultProps = {
  size: 'small',
  label: 'small'
};
PizzaPortion.propTypes = {
  size: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};
export default PizzaPortion;