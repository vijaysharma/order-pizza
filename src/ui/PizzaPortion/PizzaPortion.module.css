.PizzaSlice {
  color: #283688;
  font-size: 1rem;
  display: inline-block;
}

.PizzaSlice span {
  display: inline-block;
  margin: 0 2.5rem;
  width: 5rem;
  vertical-align: middle;
  text-align: left;
}

.SmallSlice, .MediumSlice, .LargeSlice {
  vertical-align: middle;
}

.PizzaSlice .SmallSlice, .PizzaSlice .MediumSlice, .PizzaSlice .LargeSlice {
  width: 1.5rem;
}

.SmallSlice {
  font-size: 1.25rem;
}

.MediumSlice {
  font-size: 1.5rem;
}

.LargeSlice {
  font-size: 1.75rem;
}