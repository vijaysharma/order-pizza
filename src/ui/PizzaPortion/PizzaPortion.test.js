import React from 'react';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import PizzaPortion from './PizzaPortion';

configure({adapter: new Adapter()});

describe('<Person/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PizzaPortion />);
  });

  it('should render <PizzaPortion/> component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render <PizzaPortion/> correctly', () => {
    wrapper.setProps({
      label: 'small'
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <PizzaPortion/> component with label MEDIUM', () => {
    wrapper.setProps({
      label: 'medium'
    });
    expect(wrapper.find('span').text().includes('MEDIUM')).toBe(true);
  });

  it('should render <PizzaPortion/> component with label LARGE', () => {
    wrapper.setProps({
      label: 'large'
    });
    expect(wrapper.find('span').text().includes('LARGE')).toBe(true);
  });

  it('should render <PizzaPortion/> component with icon', () => {
    expect(wrapper.find(FontAwesomeIcon)).toHaveLength(1);
  });


});