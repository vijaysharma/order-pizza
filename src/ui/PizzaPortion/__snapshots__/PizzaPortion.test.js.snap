// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Person/> should render <PizzaPortion/> correctly 1`] = `ShallowWrapper {}`;
